src/daemon/XorgDisplayServer.cpp

```diff
@@ -287,6 +290,16 @@ namespace SDDM {
             setCursor->kill();
         }
 
+        // Unlike libXcursor, xcb-util-cursor no longer looks at XCURSOR_THEME. Set the resource.
+        if (!xcursorTheme.isEmpty()) {
+            QProcess xrdbProcess;
+            xrdbProcess.setProcessEnvironment(env);
+            xrdbProcess.start(QStringLiteral("xrdb"), QStringList{QStringLiteral("-nocpp"), QStringLiteral("-merge")});
+            xrdbProcess.write(QStringLiteral("Xcursor.theme: %1").arg(xcursorTheme).toUtf8());
```

Comment:
Xcursor.size too?

Reply:
IIRC Qt actually looks at `XCURSOR_SIZE` still, but for consistency this might be better. I'll add it.

Reply:
src/gui/platform/unix/qgenericunixthemes.cpp:    static const int xCursorSize = qEnvironmentVariableIntValue("XCURSOR_SIZE");

You are right, and it doesn't read XCursor.size, don't bother.

Reply:
For consistency I added it anyway. With rootful X11, the X server state is somewhat kept until the DE, so if we leak some settings it should be consistent at least.